### PR TITLE
[Enhancement](planner&Nereids)support sql_select_limit for master

### PR DESCRIPTION
**Pull request by sohardforaname**
## Proposed changes

Issue Number: close #xxx

support sql_select_limit for original planner and Nereids.
if enable the variable
In original planner, add limit to the top planNode
In Nereids, add limit node to the top in preprocess phase. 

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — COMMENTED** [review]
add some test with cte, subquery and enable/disable pipeline engine

**Review by morrySnow — COMMENTED** [review]
add some test with cte, subquery and enable/disable pipeline engine

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run fe ut p0

**Comment by sohardforaname** [comment]
run fe ut

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/AddSqlSelectLimit.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import org.apache.doris.nereids.trees.plans.logical.LogicalCTE;
+import org.apache.doris.nereids.trees.plans.logical.LogicalLimit;
+import org.apache.doris.nereids.trees.plans.logical.LogicalPlan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalSort;
+import org.apache.doris.qe.ConnectContext;
+
+/**
+ * handle sql_select_limit and default_order_by_limit
+ */
+public class AddSqlSelectLimit extends PlanPreprocessor {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/AddSqlSelectLimit.java:33]: not use preprocessor to do this. instead, use a custom rewrite rule

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Review by starocean999 — DISMISSED** [review]

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by morrySnow — COMMENTED** [review]
change default value of sql_select_limit to -1

**Review by Kikyou1997 — APPROVED** [review]

**Pull request merged**